include/oneapi/dpl/pstl/onedpl_config.h

```diff
@@ -277,6 +277,10 @@
 #define _ONEDPL_EXCLUSIVE_SCAN_WITH_BINARY_OP_AMBIGUITY                                                                \
     (__GLIBCXX__ && __GLIBCXX__ > 20190503 && __cplusplus >= 201703L)
 
+// some algorithms in <numeric> such as 'reduce' were added since libstdc++-9.3, we
+// have to provide our own implementation if legacy libstdc++ is in use.
+#define _ONEDPL_HAS_NUMERIC_SERIAL_IMPL (__GLIBCXX__ && __GLIBCXX__ < 20200312)
```

Comment:
This is not a correct version check for libstdc++, because it only uses the release date. For example, this check will incorrectly fail for GCC 8.5 released this year. A correct check must also include some condition on the major version via `_GLIBCXX_RELEASE` macro; see e.g. line 178-179. In your case, you want to check that either the major version is less than 9, or it is equal to 9 and the build date is less than 20200312.

Reply:
Hi, @akukanov 
I have updated the macro definition to:
#define _ONEDPL_HAS_NUMERIC_SERIAL_IMPL (__GLIBCXX__ && __GLIBCXX__ < 20200312 && (_GLIBCXX_RELEASE <= 9))
Thanks very much.

Reply:
It's still incorrect. If we take GCC 8.5 (with the release date in 2021), `__GLIBCXX__ < 20200312` still evaluates as false, and so the whole condition becomes false.
This is what you should do instead:

    #define _ONEDPL_HAS_NUMERIC_SERIAL_IMPL (__GLIBCXX__ && (_GLIBCXX_RELEASE < 9 || (_GLIBCXX_RELEASE == 9 && __GLIBCXX__ < 20200312)))

Reply:
Hi, @akukanov 
The incorrect check has been fixed. I looked up libstdc++-8.5 source and it doesn't include std::reduce, so we need provide own implementation.
Thanks very much.

Reply:
Hi, @akukanov 
Could you help review again?
Thanks very much.